### ISSUE TITLE
Wrong validator result for quoted JSON

### ISSUE THREAD

**Issue by arigon** [json-iterator/go#468]
I am testing arbitrary user input for my application. I am validating the JSON input to ensure only valid JSON will be stored in the database. When using input from Javascript JSON.Stringify() it will be escaped like this:
`"{\n  \"a\": \"a\",\n\"b\n}"`

Interestingly the jsoniter.Valid() function will result in _true_ for this case which is wrong.

After unquoting the input with strconv.Unquote() it will result in _false_ which is correct.
```
{
  "a": "a",
"b
}
```

**Comment by funny-falcon** [comment]
Indeed, first version is valid JSON string. Therefore it is valid JSON.
